### ISSUE TITLE
Feature Request: Detect Informal Tag Notations (Note:, TODO:, etc.)

### ISSUE THREAD

**Issue by nijikon** [mensfeld/yard-lint#33]
## Summary

Add a new rule `Tags/InformalTagNotation` that detects informal inline notations like `Note:`, `TODO:`, `Warning:`, etc. in documentation comments and suggests converting them to proper YARD tags (`@note`, `@todo`, `@deprecated`, etc.).

## Problem

Developers often write informal notations in documentation comments:

```ruby
# Creates the index if it doesn't exist and ensure_index_exists is true.
# This ensures filterable_attributes are configured BEFORE any documents are added,
# preventing "Attribute X is not filterable" errors on first-time indexing.
#
# Note: Always waits for index creation to complete (regardless of should_wait)
# to avoid race conditions with subsequent get_settings calls.
#
# @param index_exists [Boolean]
def create_index_if_needed(index_exists)
  # ...
end
```

While readable, this informal `Note:` notation:

1. **Loses semantic meaning** - YARD doesn't recognize it as a note
2. **Inconsistent rendering** - Won't be highlighted or styled in generated documentation
3. **Poor tooling support** - IDEs and documentation generators can't parse it
4. **Reduced discoverability** - Notes buried in prose vs. structured tag sections

## Expected Behavior

The above should be flagged with a suggestion to use `@note`:

```ruby
# Creates the index if it doesn't exist and ensure_index_exists is true.
# This ensures filterable_attributes are configured BEFORE any documents are added,
# preventing "Attribute X is not filterable" errors on first-time indexing.
#
# @note Always waits for index creation to complete (regardless of should_wait)
#   to avoid race conditions with subsequent get_settings calls.
#
# @param index_exists [Boolean]
def create_index_if_needed(index_exists)
  # ...
end
```

## Patterns to Detect

| Informal Pattern | Suggested YARD Tag | Notes |
|------------------|-------------------|-------|
| `Note:`, `NOTE:`, `N.B.:` | `@note` | Common documentation notes |
| `Todo:`, `TODO:`, `FIXME:`, `XXX:` | `@todo` | Incomplete work markers |
| `Warning:`, `WARN:`, `Caution:`, `CAUTION:` | `@note` | Could also suggest custom `@warning` |
| `Deprecated:`, `DEPRECATED:` | `@deprecated` | Deprecation notices |
| `See:`, `See also:`, `SEE:` | `@see` | Cross-references |
| `Since:`, `SINCE:`, `Since version` | `@since` | Version information |
| `Author:`, `AUTHOR:` | `@author` | Authorship |
| `Version:`, `VERSION:` | `@version` | Version information |
| `Example:`, `EXAMPLE:` | `@example` | When not already in a code block |

## Configuration

```yaml
Tags/InformalTagNotation:
  Enabled: true
  Severity: convention

  # Patterns to detect and their suggested replacements
  Patterns:
    Note: '@note'
    NOTE: '@note'
    'N.B.': '@note'
    Todo: '@todo'
    TODO: '@todo'
    FIXME: '@todo'
    XXX: '@todo'
    Warning: '@note'
    WARN: '@note'
    Caution: '@note'
    CAUTION: '@note'
    Deprecated: '@deprecated'
    DEPRECATED: '@deprecated'
    See: '@see'
    'See also': '@see'
    Since: '@since'
    Author: '@author'
    Version: '@version'
    Example: '@example'
```

### Flagged (Should Use YARD Tag)

```ruby
# Note: This method is expensive and should be cached.
# TODO: Add support for batch processing.
# Warning: This will delete all records!
# Deprecated: Use #new_method instead.
# See: https://example.com/docs for more information.
# Since: version 2.0
# Author: John Doe
```

### Not Flagged (Valid)

```ruby
# @note This method is expensive and should be cached.
# @todo Add support for batch processing.
# @deprecated Use #new_method instead.
# @see https://example.com/docs
# @since 2.0
# @author John Doe

# Please note that this is different from...  (mid-sentence, not a tag pattern)
# The musical note was...  (different meaning of "note")
# See the implementation for details.  (no colon, part of sentence)
```

### Edge Cases

```ruby
# NOTE - This should probably be flagged (dash instead of colon)
# Note that this is...  (no colon - should NOT be flagged, it's prose)
# Returns a note object  (different meaning - should NOT be flagged)
```

## Message Format

```
Informal notation 'Note:' detected. Consider using the YARD tag '@note' instead
for better documentation rendering and tooling support.

  Before: # Note: Always waits for completion.
  After:  # @note Always waits for completion.
```

## Benefits

1. **Consistency** - Enforces consistent use of YARD tags across the codebase
2. **Better documentation** - YARD tags are rendered with proper styling and structure
3. **IDE support** - Tools like Solargraph, RubyMine can parse and display YARD tags
4. **Discoverability** - Notes, todos, and deprecations appear in dedicated sections
5. **Searchability** - Easy to find all `@todo` or `@deprecated` items in generated docs
